include/swift/SIL/OwnershipUtils.h

```diff
@@ -998,6 +998,26 @@ bool getAllOwnedValueIntroducers(SILValue value,
 
 OwnedValueIntroducer getSingleOwnedValueIntroducer(SILValue value);
 
+using BaseValueSet = SmallPtrSet<SILValue, 8>;
+
+/// Starting from \p initialScopeOperand, find all reborrows and their
+/// corresponding base values, and run the visitor function \p
+/// visitReborrowBaseValuePair on them. \p reborrowToBaseValuesMap is a map of
+/// reborrow phi arg and its base values that have been visited. Note
+/// that a reborrow phi, can have different base values based on different
+/// control flow paths.
```

Comment:
I don't like this API. Returning results in a map _and_ passing a callback looks like redundant.
Also, it's not possible to understand what this API does without looking (for a long time) into its implementation.

This reborrow "magic" where the base value is also passed as another (!) argument than the borrow values (If I understood correctly) needs to be documented somewhere.
@gottesmm There should be an Ownership sub-section in SIL.rst explaining this complications of re-borrowing.
At least there should be some SIL examples. Otherwise it's really hard to understand.

Reply:
I agree. I fixed the api to only pass the (phiArg, baseValue) pair to visitor now. I'll iterate once again to see if I can add helpful comments.

Would you be okay if the documentation is done in a different PR ?

Reply:
sure